api/src/main/java/com/epam/pipeline/mapper/cluster/KubernetesMapper.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017-2024 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.mapper.cluster;
+
+import com.epam.pipeline.entity.cluster.ContainerInstance;
+import com.epam.pipeline.entity.cluster.ContainerInstanceStatus;
+import com.epam.pipeline.entity.cluster.CorePodInstance;
+import com.epam.pipeline.entity.cluster.EventEntity;
+import com.epam.pipeline.entity.cluster.PodInstanceStatus;
+import io.fabric8.kubernetes.api.model.ContainerState;
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodStatus;
+import org.apache.commons.collections4.CollectionUtils;
+import org.mapstruct.Mapper;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Mapper(componentModel = "spring")
+public interface KubernetesMapper {
+
+    EventEntity mapEvent(Event event);
+
+    default CorePodInstance mapCorePod(final Pod kubePod) {
```

Comment:
if we will move status and containers in https://github.com/epam/cloud-pipeline/pull/3557/files#diff-4dfff71cd8f8757c75a2e2887c9632ff9a5b6ea9c4fe336a9b5b0fea351e9f5fR29 to PodInstance, would this method helpful in other places where we map pod object? or maybe we can reuse already existing code to map it and remove this one